base/associative.jl

```diff
@@ -121,6 +127,14 @@ julia> collect(values(a))
 """
 values(a::Associative) = ValueIterator(a)
 
+"""
+    pairs(collection)
+
+Return an iterator over `(index, value)` pairs for any
```

Comment:
Dumb question: why not having `pairs` return an iterator of `Pair`s, i.e. `index => value`.

Reply:
That is a totally reasonable question. Probably just because `zip` conveniently provides this. The `Pair` type is a bit unfortunate; we could probably get away with making `a=>b` syntax for `(a, b)`, except that pairs print more nicely for this case. I guess I should make it return `Pair`s for now.

Reply:
Having `pairs` not return `Pairs` does seem awkward.